libtreematch.la is getting put in wrapper "libs_static"

Working with head of OMPI master branch as of Fri afternoon (hash 8f19f0f01a1837b2a048b94bec9b89541ea42617) on CentOS7. This is with a fresh clone of the repo. I am able to build the branch, but cannot compile any application programs due to the following:
```
$ make add_host
mpicc -g --openmpi:linkall    add_host.c   -o add_host
/opt/hpc/build/ompi/3rd-party/treematch/libtreematch.la: file not recognized: File format not recognized
collect2: error: ld returned 1 exit status
make: *** [add_host] Error 1
```
Any ideas out there would be appreciated.

--- a/config/opal_mca.m4
+++ b/config/opal_mca.m4
@@ -725,6 +725,23 @@ AC_DEFUN([MCA_COMPONENT_COMPILE_MODE],[
     fi
 ])
 
+# OPAL_MCA_STRIP_LAFILES(output_variable(1),
+#                        input_list(2)
+#--------------------------------------------
+# Helper function to MCA_PROCESS_COMPONENT which will strip
+# any .la file entries in the LIBS list.  Used for when copying
+# a component's LIBS into WRAPPER_LIBS.
+AC_DEFUN([OPAL_MCA_STRIP_LAFILES], [
+    OPAL_VAR_SCOPE_PUSH([opal_tmp])
+
+    for arg in $2; do
+	opal_tmp=`echo $arg | awk '{print substr([$][1], length([$][1])-2) }'`
+        AS_IF([test "$opal_tmp" != ".la"],
+              [AS_IF([test -z "$$1"], [$1=$arg], [$1="$$1 $arg"])])
+    done
+
+    OPAL_VAR_SCOPE_POP
+])
 
 # MCA_PROCESS_COMPONENT(project_name(1), framework_name (2), component_name (3),
 #                        all_components_variable (4), static_components_variable (5)
@@ -806,15 +823,19 @@ AC_MSG_ERROR([*** $2 component $3 was supposed to be direct-called, but
     # into the wrapper flags.
     AS_IF([test "$8" = "static"],
           [m4_foreach(flags, [LDFLAGS, LIBS],
-                      [AS_VAR_SET_IF([$2_$3_WRAPPER_EXTRA_]flags,
+                      [m4_if(flags, [LIBS],
+                             [OPAL_MCA_STRIP_LAFILES([tmp_]flags, [$$2_$3_]flags)],
+                             [tmp_]flags[=$$2_$3_]flags)
+                       AS_VAR_SET_IF([$2_$3_WRAPPER_EXTRA_]flags,
                                      [OPAL_FLAGS_APPEND_UNIQ([mca_wrapper_extra_]m4_tolower(flags), [$$2_$3_WRAPPER_EXTRA_]flags)],
-                                     [OPAL_FLAGS_APPEND_UNIQ([mca_wrapper_extra_]m4_tolower(flags), [$$2_$3]_flags)])
+                                     [OPAL_FLAGS_APPEND_UNIQ([mca_wrapper_extra_]m4_tolower(flags), [$tmp_]flags)])
                        dnl yes, this is weird indenting, but the
                        dnl combination of m4_foreach and AS_VAR_SET_IF
                        dnl will result in the closing of one if and the
                        dnl start of the next on the same line, resulting
 		       dnl in parse errors, if this is not here.
 		       ])])
+
 
     # if needed, copy over WRAPPER_EXTRA_CPPFLAGS.  Since a configure script
     # component can never be used in a STOP_AT_FIRST framework, we

--- a/config/opal_mca.m4
+++ b/config/opal_mca.m4
@@ -1,4 +1,4 @@
-dnl -*- shell-script -*-
+dnl -*- autoconf -*-
 dnl
 dnl Copyright (c) 2004-2005 The Trustees of Indiana University and Indiana
 dnl                         University Research and Technology

--- a/config/opal_mca.m4
+++ b/config/opal_mca.m4
@@ -237,13 +237,8 @@ AC_DEFUN([OPAL_MCA],[
     MCA_PROJECT_SUBDIRS=
     MCA_PROJECT_DIST_SUBDIRS=
     m4_foreach(mca_project, [mca_project_list],
-               [# BWB: Until projects have separate configure scripts
-                # and can skip running all of ORTE, just avoid recursing
-                # into orte sub directory if orte disabled
-                if (test "mca_project" = "ompi" && test "$enable_mpi" != "no") || test "mca_project" = "opal" || test "mca_project" = "orte" || test "mca_project" = "oshmem"; then
-                   MCA_PROJECT_SUBDIRS="$MCA_PROJECT_SUBDIRS mca_project"
-                   MCA_PROJECT_DIST_SUBDIRS="$MCA_PROJECT_DIST_SUBDIRS mca_project"
-                fi
+               [MCA_PROJECT_SUBDIRS="$MCA_PROJECT_SUBDIRS mca_project"
+                MCA_PROJECT_DIST_SUBDIRS="$MCA_PROJECT_DIST_SUBDIRS mca_project"
                 MCA_CONFIGURE_PROJECT(mca_project)])
 
     AC_SUBST(MCA_PROJECT_SUBDIRS)

--- a/config/opal_mca.m4
+++ b/config/opal_mca.m4
@@ -103,12 +103,10 @@ AC_DEFUN([OPAL_MCA],[
                 type=$item
             fi
             if test -z $comp ; then
-                str="`echo DISABLE_${type}=1 | sed s/-/_/g`"
-                eval $str
+                AS_VAR_COPY([AS_TR_SH([DISABLE_$type])], [1])
                 msg="$item $msg"
             else
-                str="`echo DISABLE_${type}_${comp}=1 | sed s/-/_/g`"
-                eval $str
+                AS_VAR_COPY([AS_TR_SH([DISABLE_$type_$comp])], [1])
                 msg="$item $msg"
             fi
         done
@@ -151,8 +149,7 @@ AC_DEFUN([OPAL_MCA],[
                 AC_MSG_ERROR([*** The enable-mca-direct flag requires a
 *** list of type-component pairs.  Invalid input detected.])
             else
-                str="`echo DIRECT_$type=$comp | sed s/-/_/g`"
-                eval $str
+                AS_VAR_COPY([AS_TR_SH([DIRECT_$type])], [AS_TR_SH([$comp])])
                 msg="$item $msg"
             fi
         done
@@ -187,8 +184,7 @@ AC_DEFUN([OPAL_MCA],[
         IFS="${IFS}$PATH_SEPARATOR,"
         msg=
         for item in $enable_mca_dso; do
-            str="`echo DSO_$item=1 | sed s/-/_/g`"
-            eval $str
+            AS_VAR_COPY([AS_TR_SH([DSO_$item])], [1])
             msg="$item $msg"
         done
         IFS="$ifs_save"
@@ -216,8 +212,7 @@ AC_DEFUN([OPAL_MCA],[
         IFS="${IFS}$PATH_SEPARATOR,"
         msg=
         for item in $enable_mca_static; do
-            str="`echo STATIC_$item=1 | sed s/-/_/g`"
-            eval $str
+            AS_VAR_COPY([AS_TR_SH([STATIC_$item])], [1])
             msg="$item $msg"
         done
         IFS="$ifs_save"
@@ -698,16 +693,10 @@ AC_DEFUN([MCA_CONFIGURE_ALL_CONFIG_COMPONENTS],[
 #   NOTE: component_name may not be determined until runtime....
 AC_DEFUN([MCA_COMPONENT_COMPILE_MODE],[
     SHARED_FRAMEWORK="$DSO_$2"
-    AS_LITERAL_IF([$3],
-        [SHARED_COMPONENT="$DSO_$2_$3"],
-        [str="SHARED_COMPONENT=\$DSO_$2_$3"
-         eval $str])
+    AS_VAR_COPY([SHARED_COMPONENT], [DSO_$2_$3])
 
     STATIC_FRAMEWORK="$STATIC_$2"
-    AS_LITERAL_IF([$3],
-        [STATIC_COMPONENT="$STATIC_$2_$3"],
-        [str="STATIC_COMPONENT=\$STATIC_$2_$3"
-         eval $str])
+    AS_VAR_COPY([STATIC_COMPONENT], [DSO_$2_$3])
 
     # Look for the most specific specifier between static/dso.  If
     # there is a tie (either neither or both specified), prefer
@@ -807,26 +796,25 @@ AC_MSG_ERROR([*** $2 component $3 was supposed to be direct-called, but
         fi
     fi
 
-    # if the component is building, add it's WRAPPER_EXTRA_LDFLAGS and
-    # WRAPPER_EXTRA_LIBS.  If the component doesn't specify it's
-    # WRAPPER_EXTRA_LIBS and WRAPPER_EXTRA_LDFLAGS, try using LDFLAGS and LIBS if
-    # component didn't have it's own configure script (in which case,
-    # we know it didn't set LDFLAGS and LIBS because it can't) Don't
-    # have to do this if the component is building dynamically,
-    # because it will link against these (without a dependency from
-    # libmpi.so to these flags)
-    if test "$8" = "static"; then
-        AS_LITERAL_IF([$3],
-            [m4_foreach(flags, [LDFLAGS, LIBS],
-                    [AS_IF([test "$$2_$3_WRAPPER_EXTRA_]flags[" = ""],
-                           [OPAL_FLAGS_APPEND_UNIQ([mca_wrapper_extra_]m4_tolower(flags), [$$2_$3_]flags)],
-                           [OPAL_FLAGS_APPEND_UNIQ([mca_wrapper_extra_]m4_tolower(flags), [$$2_$3_WRAPPER_EXTRA_]flags)])
-                        ])],
-            [m4_foreach(flags, [LDFLAGS, LIBS],
-                    [[str="line=\$$2_$3_WRAPPER_EXTRA_]flags["]
-                      eval "$str"
-                      OPAL_FLAGS_APPEND_UNIQ([mca_wrapper_extra_]m4_tolower(flags), [$line])])])
-    fi
+    # If a component is building static, we need to provide LDFLAGS
+    # and LIBS configuration to the wrapper compiler, so that it can
+    # provide them for the final link of the application.  Components
+    # can explicitly set <framework>_<component>_WRAPPER_EXTRA_<flag>
+    # for either LDFLAGS or LIBS, for cases where the component wants
+    # to explicitly manage that behavior.  If the full variable is not
+    # defined, this macro will copy <framework>_<component>_<flag>
+    # into the wrapper flags.
+    AS_IF([test "$8" = "static"],
+          [m4_foreach(flags, [LDFLAGS, LIBS],
+                      [AS_VAR_SET_IF([$2_$3_WRAPPER_EXTRA_]flags,
+                                     [OPAL_FLAGS_APPEND_UNIQ([mca_wrapper_extra_]m4_tolower(flags), [$$2_$3_WRAPPER_EXTRA_]flags)],
+                                     [OPAL_FLAGS_APPEND_UNIQ([mca_wrapper_extra_]m4_tolower(flags), [$$2_$3]_flags)])
+                       dnl yes, this is weird indenting, but the
+                       dnl combination of m4_foreach and AS_VAR_SET_IF
+                       dnl will result in the closing of one if and the
+                       dnl start of the next on the same line, resulting
+		       dnl in parse errors, if this is not here.
+		       ])])
 
     # if needed, copy over WRAPPER_EXTRA_CPPFLAGS.  Since a configure script
     # component can never be used in a STOP_AT_FIRST framework, we
@@ -925,13 +913,7 @@ AC_DEFUN([MCA_COMPONENT_BUILD_CHECK],[
 
     # if we were explicitly disabled, don't build :)
     AS_IF([test "$DISABLE_$2" = "1"], [want_component=0])
-    AS_LITERAL_IF([$3],
-        [AS_IF([test "$DISABLE_$2_$3" = "1"], [want_component=0])],
-        [str="DISABLED_COMPONENT_CHECK=\$DISABLE_$2_$3"
-         eval $str
-         if test "$DISABLED_COMPONENT_CHECK" = "1" ; then
-             want_component=0
-         fi])
+    AS_VAR_IF([DISABLE_$2_$3], [1], [want_component = 0])
 
     AS_IF([test "$want_component" = "1"], [$4], [$5])
 ])

--- a/config/opal_setup_wrappers.m4
+++ b/config/opal_setup_wrappers.m4
@@ -383,54 +383,6 @@ AC_DEFUN([OPAL_SETUP_WRAPPER_FINAL],[
        AC_MSG_RESULT([$OPAL_WRAPPER_EXTRA_LIBS])
     ])
 
-    m4_ifdef([project_orte], [
-       AC_MSG_CHECKING([for ORTE CPPFLAGS])
-       if test "$WANT_INSTALL_HEADERS" = "1" ; then
-           ORTE_WRAPPER_EXTRA_CPPFLAGS='-I${includedir}/openmpi'
-       fi
-       ORTE_WRAPPER_EXTRA_CPPFLAGS="$ORTE_WRAPPER_EXTRA_CPPFLAGS $orte_mca_wrapper_extra_cppflags $wrapper_extra_cppflags $with_wrapper_cppflags"
-       AC_SUBST([ORTE_WRAPPER_EXTRA_CPPFLAGS])
-       AC_MSG_RESULT([$ORTE_WRAPPER_EXTRA_CPPFLAGS])
-
-       AC_MSG_CHECKING([for ORTE CFLAGS])
-       ORTE_WRAPPER_EXTRA_CFLAGS="$wrapper_extra_cflags $with_wrapper_cflags"
-       AC_SUBST([ORTE_WRAPPER_EXTRA_CFLAGS])
-       AC_MSG_RESULT([$ORTE_WRAPPER_EXTRA_CFLAGS])
-
-       AC_MSG_CHECKING([for ORTE CFLAGS_PREFIX])
-       ORTE_WRAPPER_EXTRA_CFLAGS_PREFIX="$with_wrapper_cflags_prefix"
-       AC_SUBST([ORTE_WRAPPER_EXTRA_CFLAGS_PREFIX])
-       AC_MSG_RESULT([$ORTE_WRAPPER_EXTRA_CFLAGS_PREFIX])
-
-       AC_MSG_CHECKING([for ORTE LDFLAGS])
-       ORTE_WRAPPER_EXTRA_LDFLAGS="$orte_mca_wrapper_extra_ldflags $wrapper_extra_ldflags $with_wrapper_ldflags"
-       RPATHIFY_LDFLAGS([ORTE_WRAPPER_EXTRA_LDFLAGS])
-       AC_SUBST([ORTE_WRAPPER_EXTRA_LDFLAGS])
-       AC_MSG_RESULT([$ORTE_WRAPPER_EXTRA_LDFLAGS])
-
-       # Convert @{libdir} to ${libdir} for pkg-config
-       _OPAL_SETUP_WRAPPER_FINAL_PKGCONFIG([ORTE])
-
-       AC_MSG_CHECKING([for ORTE LIBS])
-       ORTE_WRAPPER_EXTRA_LIBS="$orte_mca_wrapper_extra_libs"
-       OPAL_FLAGS_APPEND_UNIQ([ORTE_WRAPPER_EXTRA_LIBS], [$wrapper_extra_libs])
-       ORTE_WRAPPER_EXTRA_LIBS="$ORTE_WRAPPER_EXTRA_LIBS $with_wrapper_libs"
-       AC_SUBST([ORTE_WRAPPER_EXTRA_LIBS])
-       AC_MSG_RESULT([$ORTE_WRAPPER_EXTRA_LIBS])
-
-       m4_ifdef([project_ompi], [], [
-          # these are used by orte_info/ompi_info (yes, they are named poorly)
-          AC_DEFINE_UNQUOTED(WRAPPER_EXTRA_CFLAGS, "$ORTE_WRAPPER_EXTRA_CFLAGS",
-              [Additional CFLAGS to pass through the wrapper compilers])
-          AC_DEFINE_UNQUOTED(WRAPPER_EXTRA_CFLAGS_PREFIX, "$ORTE_WRAPPER_EXTRA_CFLAGS_PREFIX",
-              [Additional CFLAGS_PREFIX to pass through the wrapper compilers])
-          AC_DEFINE_UNQUOTED(WRAPPER_EXTRA_LDFLAGS, "$ORTE_WRAPPER_EXTRA_LDFLAGS",
-              [Additional LDFLAGS to pass through the wrapper compilers])
-          AC_DEFINE_UNQUOTED(WRAPPER_EXTRA_LIBS, "$ORTE_WRAPPER_EXTRA_LIBS",
-              [Additional LIBS to pass through the wrapper compilers])
-       ])
-    ])
-
     m4_ifdef([project_ompi], [
        AC_MSG_CHECKING([for OMPI CPPFLAGS])
        if test "$WANT_INSTALL_HEADERS" = "1" ; then

